Data race in TestVSphereLoginWithCaCert

**What happened**:
Data race in tests in an unrelated PR https://github.com/kubernetes/kubernetes/pull/64654
Logs https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/64654/pull-kubernetes-bazel-test/51222/
```
Executing tests from //pkg/cloudprovider/providers/vsphere:go_default_test
-----------------------------------------------------------------------------
E0702 06:51:23.931693      15 credentialmanager.go:91] Cannot get secret vsconf in namespace kube-system. error: "secret \"vsconf\" not found"
E0702 06:51:24.113654      15 credentialmanager.go:91] Cannot get secret vsconf in namespace kube-system. error: "secret \"vsconf\" not found"
E0702 06:51:24.115151      15 credentialmanager.go:79] credentials not found for server "0.0.0.0"
E0702 06:51:24.115838      15 credentialmanager.go:79] credentials not found for server "1.1.1.1"
E0702 06:51:24.116971      15 credentialmanager.go:153] Unknown secret key 10.20.30.40.usernam
E0702 06:51:24.117098      15 credentialmanager.go:159] Username/Password is missing for server 10.20.30.40
E0702 06:51:24.117189      15 credentialmanager.go:159] Username/Password is missing for server 10.20.30.40
E0702 06:51:24.117270      15 credentialmanager.go:153] Unknown secret key 10.20.30.40
E0702 06:51:24.192449      15 connection.go:142] No active session, cannot logout
E0702 06:51:24.260716      15 connection.go:142] No active session, cannot logout
E0702 06:51:24.366640      15 connection.go:142] No active session, cannot logout
==================
WARNING: DATA RACE
Write at 0x00c42013d100 by goroutine 80:
  k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere/vclib.(*VSphereConnection).Connect()
      pkg/cloudprovider/providers/vsphere/vclib/connection.go:61 +0x3ff
  k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere.TestVSphereLoginWithCaCert()
      pkg/cloudprovider/providers/vsphere/vsphere_test.go:310 +0x717
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d

Previous read at 0x00c42013d100 by goroutine 5:
  k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere.logout()
      pkg/cloudprovider/providers/vsphere/vsphere.go:513 +0x10b

Goroutine 80 (running) created at:
  testing.(*T).Run()
      GOROOT/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      GOROOT/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d
  testing.runTests()
      GOROOT/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      GOROOT/src/testing/testing.go:978 +0x2cd
  main.main()
      bazel-out/k8-fastbuild/bin/pkg/cloudprovider/providers/vsphere/linux_amd64_race_stripped/go_default_test~/testmain.go:89 +0x28e

Goroutine 5 (running) created at:
  runtime.createfing()
      GOROOT/src/runtime/mfinal.go:156 +0x61
  os.NewFile()
      GOROOT/src/os/file_unix.go:79 +0x55
  os.init()
      GOROOT/src/os/file.go:58 +0x418
  fmt.init()
      <autogenerated>:1 +0xa0
  flag.init()
      <autogenerated>:1 +0x96
  main.init()
      <autogenerated>:1 +0x96
==================
E0702 06:51:24.637923      15 connection.go:142] No active session, cannot logout
--- FAIL: TestVSphereLoginWithCaCert (0.27s)
	testing.go:730: race detected during execution of test
E0702 06:51:24.646383      15 vsphere.go:312] Global.User is empty!
E0702 06:51:24.647262      15 vsphere.go:316] Global.Password is empty!
E0702 06:51:24.648055      15 vsphere.go:312] Global.User is empty!
FAIL
```
/kind bug
/area platform/vsphere 
/area cloudprovider
/sig cloud-provider

--- a/pkg/cloudprovider/providers/vsphere/vclib/connection.go
+++ b/pkg/cloudprovider/providers/vsphere/vclib/connection.go
@@ -131,7 +131,14 @@ func (connection *VSphereConnection) login(ctx context.Context, client *vim25.Cl
 
 // Logout calls SessionManager.Logout for the given connection.
 func (connection *VSphereConnection) Logout(ctx context.Context) {
-	m := session.NewManager(connection.Client)
+	clientLock.Lock()
+	c := connection.Client
+	clientLock.Unlock()
+	if c == nil {
+		return
+	}
+
+	m := session.NewManager(c)
 
 	hasActiveSession, err := m.SessionIsActive(ctx)
 	if err != nil {

--- a/pkg/cloudprovider/providers/vsphere/vsphere.go
+++ b/pkg/cloudprovider/providers/vsphere/vsphere.go
@@ -510,11 +510,8 @@ func buildVSphereFromConfig(cfg VSphereConfig) (*VSphere, error) {
 
 func logout(vs *VSphere) {
 	for _, vsphereIns := range vs.vsphereInstanceMap {
-		if vsphereIns.conn.Client != nil {
-			vsphereIns.conn.Logout(context.TODO())
-		}
+		vsphereIns.conn.Logout(context.TODO())
 	}
-
 }
 
 // Instances returns an implementation of Instances for vSphere.

